src/unix/udp.c

```diff
@@ -507,6 +507,28 @@ static int uv__set_reuse(int fd) {
   return 0;
 }
 
+/*
+ * The Linux kernel suppresses some ICMP error messages by default for UDP
+ * sockets. Setting IP_RECVERR/IPV6_RECVERR on the socket enables full ICMP
+ * error reporting, hopefully resulting in faster failover to working name
+ * servers.
+ */
+static int uv__set_recverr(int fd, sa_family_t ss_family) {
+#if defined(__linux__)
+  int yes;
+
+  yes = 1;
+  if (ss_family == AF_INET) {
+    if (setsockopt(fd, IPPROTO_IP, IP_RECVERR, &yes, sizeof(yes)))
+      return UV__ERR(errno);
+  } else if (ss_family == AF_INET6) {
+    if (setsockopt(fd, IPPROTO_IPV6, IPV6_RECVERR, &yes, sizeof(yes)))
+       return UV__ERR(errno);
+  }
+#endif
+  return 0;
```

Comment:
```suggestion
  return 0;
#endif
  return UV_ENOSYS;
```
Unless there's a compelling reason to keep it a no-op on non-linux.

Reply:
That change would require every client application to apply the `UV_UDP_RECVERR` selectively, so any client application using `UV_UDP_RECVERR` would have to do:
```
#ifdef __linux__
uv_bind(..., UV_UDP_RECVERR);
#else
uv_bind(..., 0);
#endif
```

I don't think this is desirable.

Reply:
Returning `UV_ENOSYS` for unsupported features is libuv policy but let's see what @libuv/collaborators think.

Reply:
While it's true that's the usual policy, there are some exceptions such in some [process flags](http://docs.libuv.org/en/v1.x/process.html#c.uv_process_flags) that are windows-only and silently ignored in other platforms. Maybe renaming it to `UV_UDP_LINUX_RECVERR` can be an acceptable solution.

Reply:
Good point and good middle ground. What do you think, @oerdnj?

Reply:
Ok, fair enough.

Reply:
Renamed in e714fe79